### PR TITLE
Update to restraints

### LINKED ISSUE #25802: IPCs can eat Twimst cuffs
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1620

### Issue Description

I put twimst cuffs on myself as an IPC and then when breaking free, ate the cuffs without a mouth.

### What did you expect to happen?

I expected to just attempt to break out and fail probably or there being a slightly longer timer than 0 seconds.

### What happened instead?

I ate the cuffs in 0 seconds.

### Why is this bad/What are the consequences?

IPCs should not be able to eat, they have no mouth!

### Steps to reproduce the issue.

buy twimsts, turn them into cuffs, ziptie self with cuffs, and then resist to eat them.

### When did the problem start happening?

I have no clue.

### Extra information

_No response_

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by FaronCD**
## What Does This PR Do

Updates restraint code to fix #25802 

## Why It's Good For The Game

Updated code is good. IPC's cant eat food, now IPC's break twimst cuffs

## Testing

It was tested on a localhost.

## Changelog
:cl: EOD501
fix: IPC's now break twimst cuffs instead of eating them
/:cl:

### CODE REVIEW

**Comment by Scribble-Sheep** [comment]
Does this also apply to species with a mouth, but who're muzzled? Could see a similar "eating through duct tape" bug.

File: code/game/objects/items/weapons/handcuffs.dm [hunk 1]
```diff
@@ -4,6 +4,12 @@
 	icon = 'icons/obj/restraints.dmi'
 	var/cuffed_state = "handcuff"
 
+/obj/item/restraints/finish_resist_restraints(mob/living/carbon/user, break_cuffs, silent) //you only want to break hand or legcuffs, not any type of restraint
```
Comment (1080pCat) [inline code/game/objects/items/weapons/handcuffs.dm:7]:
```suggestion
/obj/item/restraints/finish_resist_restraints(mob/living/carbon/user, break_cuffs, silent) // you only want to break hand or legcuffs, not any type of restraint
```

File: code/game/objects/items/weapons/handcuffs.dm [hunk 2]
```diff
@@ -229,3 +235,19 @@
 	throwforce = 0
 	breakouttime = 0
 	cuffsound = 'sound/weapons/cablecuff.ogg'
+
+/obj/item/restraints/handcuffs/twimsts/finish_resist_restraints(mob/living/carbon/user, break_cuffs)
+	if(ishuman(user))
+		var/mob/living/carbon/human/human_user = user
+		if(!human_user.check_has_mouth()) //I have no mouth but I must eat twimsts
```
Comment (1080pCat) [inline code/game/objects/items/weapons/handcuffs.dm:242]:
```suggestion
		if(!human_user.check_has_mouth()) // I have no mouth but I must eat twimsts
```

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -927,3 +927,25 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
 
 /obj/item/proc/should_stack_with(obj/item/other)
 	return type == other.type && name == other.name
+
+/obj/item/proc/attempt_resist_restraints(mob/living/carbon/user, break_cuffs, effective_breakout_time, silent)
+	if(effective_breakout_time)
+		if(!silent)
+			user.visible_message("<span class='warning'>[user] attempts to [break_cuffs ? "break" : "remove"] [src]!</span>", "<span class='notice'>You attempt to [break_cuffs ? "break" : "remove"] [src]...")
```
Comment (Burzah) [inline code/game/objects/items.dm:934]:
```suggestion
			user.visible_message("<span class='warning'>[user] attempts to [break_cuffs ? "break" : "remove"] [src]!</span>", "<span class='notice'>You attempt to [break_cuffs ? "break" : "remove"] [src]...</span>")
```

File: code/modules/mob/living/carbon/alien/humanoid/humanoid.dm [hunk 4]
```diff
@@ -78,7 +78,7 @@ GLOBAL_LIST_INIT(strippable_alien_humanoid_items, create_strippable_list(list(
 
 /mob/living/carbon/alien/humanoid/cuff_resist(obj/item/I)
 	playsound(src, 'sound/voice/hiss5.ogg', 40, 1, 1)  //Alien roars when starting to break free
```
Comment (Burzah) [inline code/modules/mob/living/carbon/alien/humanoid/humanoid.dm:80]:
```suggestion
	playsound(src, 'sound/voice/hiss5.ogg', 40, TRUE, 1)  //Alien roars when starting to break free
```
While we are here.

File: code/game/gamemodes/cult/blood_magic.dm [hunk 5]
```diff
@@ -616,18 +616,20 @@
 
 
 /// For the shackling spell
-/obj/item/restraints/handcuffs/energy/cult
+/obj/item/restraints/handcuffs/cult
 	name = "shadow shackles"
 	desc = "Shackles that bind the wrists with sinister magic."
-	trashtype = /obj/item/restraints/handcuffs/energy/used
+	icon_state = "cablecuff"
+	breakouttime = 450
```
Comment (chuga-git) [inline code/game/gamemodes/cult/blood_magic.dm:623]:
```suggestion
	breakouttime = 45 SECONDS
```

File: code/game/objects/items.dm [hunk 6]
```diff
@@ -927,3 +927,25 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
 
 /obj/item/proc/should_stack_with(obj/item/other)
 	return type == other.type && name == other.name
+
+/obj/item/proc/attempt_resist_restraints(mob/living/carbon/user, break_cuffs, effective_breakout_time, silent)
```
Comment (Sirryan2002) [inline code/game/objects/items.dm:931]: why isn't this scoped to `/obj/item/restraints` ???

File: code/game/objects/items.dm [hunk 7]
```diff
@@ -927,3 +927,25 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
...
+		to_chat(user, "<span class='warning'>You fail to [break_cuffs ? "break" : "remove"] [src]!</span>")
+		return
+
+	user.remove_status_effect(STATUS_EFFECT_REMOVE_CUFFS)
+	if(loc != user || user.buckled)
+		return
+
+	finish_resist_restraints(user, break_cuffs)
+
+/obj/item/proc/finish_resist_restraints(mob/living/carbon/user, break_cuffs, silent)
```
Comment (Sirryan2002) [inline code/game/objects/items.dm:948]: same here, why not scoped to `/obj/item/restraints` ?

**Review by Sirryan2002 — CHANGES_REQUESTED** [review]

File: code/game/objects/items.dm (hunk 6 above)
Comment (Sirryan2002) [reply code/game/objects/items.dm:931]: would you just duplicate the code on the straight_jacket? Duplicate code isn't ideal, but this also really doesn't need to be scoped to such a low level type

File: code/modules/mob/living/carbon/alien/humanoid/humanoid.dm [hunk 8]
```diff
@@ -76,9 +76,10 @@ GLOBAL_LIST_INIT(strippable_alien_humanoid_items, create_strippable_list(list(
 	. = ..()
 	. += alien_movement_delay
 
-/mob/living/carbon/alien/humanoid/cuff_resist(obj/item/I)
+/mob/living/carbon/alien/humanoid/resist_restraints(attempt_breaking)
 	playsound(src, 'sound/voice/hiss5.ogg', 40, 1, 1)  //Alien roars when starting to break free
-	..(I, cuff_break = 1)
+	attempt_breaking = TRUE
+	. = ..()
```
Comment (Sirryan2002) [inline code/modules/mob/living/carbon/alien/humanoid/humanoid.dm:82]:
```suggestion
	return ..()
```

File: code/modules/mob/living/carbon/human/human_mob.dm [hunk 9]
```diff
@@ -988,20 +988,11 @@
...
-		if(..())
-			unEquip(I)
-
-/mob/living/carbon/human/resist_restraints()
-	if(wear_suit && wear_suit.breakouttime)
-		cuff_resist(wear_suit)
-	else
-		..()
+		attempt_breaking = TRUE
+	. = ..()
```
Comment (Sirryan2002) [inline code/modules/mob/living/carbon/human/human_mob.dm:995]:
```suggestion
	return ..()
```

**Review by Sirryan2002 — APPROVED** [review]

**Review by Sirryan2002 — APPROVED** [review]

**Review by lewcc — APPROVED** [review]

**Pull request merged**